src/color/Color.ts

```diff
@@ -317,4 +320,32 @@ export class Color {
       return [r, g, b, a / 255];
     }
   }
+
+  /**
+   * Converts a string that could be any angle notation (50deg, 0.5turn, 2rad)
+   * into degrees without the 'deg' suffix
+   * @static
+   * @memberOf Color
+   * @param {String} value ex: 0deg, 0.5turn, 2rad
+   * @return {String} numeric string in degrees, or '0' for improper input
+   */
+  static parseAngletoDegrees(value: string): string {
+    const lowercase = value.toLowerCase();
+    if (lowercase.indexOf('deg') > -1) {
+      return lowercase.replace('deg', '');
+    }
+
+    if (lowercase.indexOf('rad') > -1) {
+      const numeric = parseFloat(lowercase.replace('rad', ''));
+      return numeric === undefined ? '0' : radiansToDegrees(numeric).toString();
+    }
+
+    if (lowercase.indexOf('turn') > -1) {
+      const numeric = parseFloat(lowercase.replace('turn', ''));
+      return numeric === undefined ? '0' : (numeric * 360).toString();
+    }
+
+    // Value is probably just a number already in degrees eg '50'
+    return lowercase;
+  }
```

Comment:
we are doing parseFloat with the return of this function.
<img width="214" alt="image" src="https://github.com/fabricjs/fabric.js/assets/1194048/680ffb5c-76ba-4077-8028-57eb03d95d4a">

parse float will already exclude strings, so all the replace ('rad') replace('turn') are unnecessary.

Have this function return a number and do a single parseFloat after lowerCase and have it return numeric.

```ts
  static parseAngletoDegrees(value: string): number {
    const lowercase = value.toLowerCase();
    const numeric = parseFloat(lowercase);
    
    if (lowercase.contains('rad')) {
      const numeric = parseFloat(lowercase.replace('rad', ''));
      return radiansToDegrees(numeric);
    }

    if (lowercase.contains('turn')) {
      const numeric = parseFloat(lowercase.replace('turn', ''));
      return numeric * 360;
    }
    return numeric;
  }
```

so the code is less

Reply:
good thinking